frontend/src/auth/pages/RegisterPage.tsx

```diff
@@ -1,3 +1,32 @@
+import { z } from 'zod'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+
+const SignUpSchema = z.object({
+  email: z.string().email(),
+  password: z.string().min(3).max(20),
+})
+type SignUpSchemaType = z.infer<typeof SignUpSchema>
+
 export function RegisterPage() {
-  return <div>Register</div>
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<SignUpSchemaType>({ resolver: zodResolver(SignUpSchema) })
```

Comment:
u can make custom hook to not dublicate code on every form:

import type z from "zod";
import { useForm, UseFormProps } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";

export default function useZodForm<TSchema extends z.Schema>(
	schema: TSchema,
	options?: Omit<UseFormProps<z.infer<TSchema>>, "resolver">
) {
	return useForm({ ...options, resolver: zodResolver(schema) });
}

and use something like that
![image](https://github.com/user-attachments/assets/7cfd9f62-fe2a-4dba-a249-494b4e0a8455)